R/input-check-helpers.R

```diff
@@ -1,114 +1,159 @@
-#' predictive samples, n (number of rows) being the number of data points and
-#' N (number of columns) the number of Monte Carlo samples
-#' @param type character, one of "continuous" (default), "integer" or "binary" that
-#' defines the type of the forecast
-#' @param class character, either "vector" (default) or "matrix" that determines the
-#' class the input has to correspond to
-#' @inheritParams ae_median_sample
-#' @return NULL
+#' @description Helper function to check whether the input is suitable for
+#' scoring.
+#' @param observed Input to be checked. Should be a vector with the
+#' observed values of size n
+#' @param predicted Input to be checked. Should be nxN matrix of predictive
+#' samples, n (number of rows) being the number of data points and N (number of
+#' columns) the number of samples per forecast.
+#' If `observed` is just a single number, then predicted values can just be a
+#' vector of size N.
+#' @importFrom checkmate assert assert_numeric check_matrix
+#' @return Returns NULL invisibly if the check was successful and throws an
+#' error otherwise.
 #' @keywords internal
 
-check_predictions <- function(predictions,
-                              true_values = NULL,
-                              type = c("continuous", "integer", "binary"),
-                              class = c("vector", "matrix")) {
-  type <- match.arg(type)
-  class <- match.arg(class)
+check_input_sample <- function(observed, predicted) {
+  # things that need to be checked
+  # - observed and predicted need to be numeric
+  # - observed needs to be a scalar or a vector of size n x 1
+  # - predicted needs to be a vector or a matrix of size n x N
 
-  if (missing(predictions)) {
-    stop("argument 'predictions' missing")
-  }
+  assert_numeric(observed, min.len = 1)
+  n_obs <- length(observed)
 
-  if (class == "vector") {
-    if (!is.vector(predictions)) {
-      msg <- sprintf(
-        "'predictions' should be a vector. Instead `%s` was found",
-        class(predictions)[1]
-      )
-      stop(msg)
-    }
-    if (!is.null(true_values) && length(predictions) != length(true_values)) {
-      msg <- sprintf(
-        "Mismatch: 'true_values' has length `%s`, but 'predictions' has length `%s`.", # nolint
-        length(true_values), length(predictions)
-      )
-      stop(msg)
-    }
-  }
-
-  if (class == "matrix") {
-    if (!is.matrix(predictions)) {
-      msg <- sprintf(
-        "'predictions' should be a matrix. Instead `%s` was found",
-        class(predictions[1])
-      )
-      stop(msg)
-    }
-    if (!is.null(true_values) && nrow(predictions) != length(true_values)) {
-      msg <- sprintf(
-        "Mismatch: 'true_values' has length `%s`, but 'predictions' has `%s` rows.",
-        length(true_values), nrow(predictions)
+  if (n_obs == 1) {
+    assert(
+      # allow one of two options
+      check_numeric_vector(predicted, min.len = 1),
+      check_matrix(predicted, mode = "numeric")
       )
```

Comment:
this comma is one tab too many - have you run a linter on this?

Reply:
Hm.. Maybe I fixed it in the newest version. Linter doesn't complain anymore at least...